src/silo/query_engine/actions/fasta_aligned.cpp

```diff
@@ -59,21 +59,19 @@ std::string reconstructSequence(
       reconstructed_sequence[position_id] = SymbolType::symbolToChar(symbol);
    }
 
-   tbb::
-      parallel_for(
-         tbb::blocked_range<size_t>(0, sequence_store.positions.size()),
-         [&](const auto local) {
-            for (auto position_id = local.begin(); position_id != local.end(); position_id++) {
-               const Position<SymbolType>& position = sequence_store.positions.at(position_id);
-               for (const auto symbol : SymbolType::SYMBOLS) {
-                  if (symbol != position.symbol_whose_bitmap_is_flipped &&
-                      position.bitmaps.at(symbol).contains(sequence_id)) {
-                     reconstructed_sequence[position_id] = SymbolType::symbolToChar(symbol);
-                  }
+   tbb::parallel_for(
+      tbb::blocked_range<size_t>(0, sequence_store.positions.size()),
+      [&](const auto local) {
+         for (auto position_id = local.begin(); position_id != local.end(); position_id++) {
+            const Position<SymbolType>& position = sequence_store.positions.at(position_id);
+            for (const auto symbol : SymbolType::SYMBOLS) {
+               if (!position.isSymbolFlipped(symbol) && position.getBitmap(symbol)->contains(sequence_id)) {
```

Comment:
Why don't we need to account for deleted symbols here?

Reply:
True, we can do that. It is not incorrect, but it is a small optimisation as we would not need to call the bitmap::contains function (which will always return false)

Reply:
Yes, exactly. I think saying that we start with the reference sequence **of the partition** is a good way to put it. That is, we change the reference sequence with regard to all members of `indexing_differences_to_reference_sequence`. This then changed all positions where the optimized symbol is different from the reference sequence we started from.